securesystemslib/metadata.py

```diff
@@ -94,3 +90,70 @@ def pae(self) -> bytes:
             len(self.payload),
             self.payload,
         )
+
+    def sign(self, signer: Signer) -> Signature:
+        """Sign the payload and create the signature.
+
+        Arguments:
+            signer: A "Signer" class instance.
+
+        Returns:
+            A "Signature" instance.
+        """
+
+        signature = signer.sign(self.pae())
+        self.signatures.append(signature)
+
+        return signature
+
+    def verify(self, keys: List[Key], threshold: int) -> Dict[str, Key]:
+        """Verify the payload with the provided Keys.
+
+        Arguments:
+            keys: A list of "Key" class instance that is a public key to verify
+                the signatures.
+            threshold: Number of signatures needed to pass the verification.
+
+        Raises:
+            ValueError: If "threshold" is not valid.
+            SignatureVerificationError: If the amount of "recognized_signers" is less
+                than provided threshold.
+
+        Returns:
+            accepted_keys: A dict of unique public keys.
+        """
+
+        accepted_keys = {}
+        pae = self.pae()
+
+        # checks for threshold value.
+        if threshold <= 0:
+            raise ValueError("Threshold must be greater than 0")
+
+        if threshold > len(keys):
+            raise ValueError("Amount of keys must be greater than threshold")
```

Comment:
The error message is still not accurate. The number of passed keys can also be equal to the threshold.

Here's what I meant by [logical operator and error message should align](https://github.com/in-toto/securesystemslib/pull/2#discussion_r914686778)
```python
if len(keys) < threshold:
  raise ValueError("Number of keys can't be less than threshold")
````
Maybe it's just me, but I would argue that one sees immediately that condition and error message mean the same thing.